src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1124,20 +1124,97 @@ protected function xmlToProps($xml)
+                if ($name === 'SV:PROPERTY' && $lastPropertyName) {
+                    switch ($lastPropertyType) {
+                        case PropertyType::BINARY:
+                            $data->{':' . $lastPropertyName} = $lastPropertyMultiValued ? $currentValues : $currentValues[0];
+                            break;
+                        default:
+                            $data->{$lastPropertyName} = $lastPropertyMultiValued ? $currentValues : $currentValues[0];
+                            $data->{':' . $lastPropertyName} = $lastPropertyType;
+                            break;
+                    }
+
+                    $currentValues = [];
+                    $lastPropertyName = null;
+                    $lastPropertyType = null;
+                    $lastPropertyMultiValued = null;
+                } elseif ($name === 'SV:VALUE' && $lastPropertyName) {
+                    switch ($lastPropertyType) {
+                        case PropertyType::NAME:
+                        case PropertyType::URI:
+                        case PropertyType::WEAKREFERENCE:
+                        case PropertyType::REFERENCE:
+                        case PropertyType::PATH:
+                        case PropertyType::DECIMAL:
+                        case PropertyType::STRING:
+                            $currentValues[] = $currentValue;
+                            break;
+                        case PropertyType::BOOLEAN:
+                            $currentValues[] = (bool)$currentValue;
+                            break;
+                        case PropertyType::LONG:
+                            $currentValues[] = (int)$currentValue;
+                            break;
+                        case PropertyType::BINARY:
+                            $currentValues[] = (int)$currentValue;
+                            break;
+                        case PropertyType::DATE:
+                            $date = $currentValue;
+                            if ($date) {
+                                $date = new DateTime($date);
+                                $date->setTimezone(new DateTimeZone(date_default_timezone_get()));
+                                // Jackalope expects a string, might make sense to refactor to allow DateTime instances too
+                                $date = $this->valueConverter->convertType($date, PropertyType::STRING);
+                            }
+                            $currentValues[] = $date;
+                            break;
+                        case PropertyType::DOUBLE:
+                            $currentValues[] = (double)$currentValue;
+                            break;
+                        default:
+                            throw new InvalidArgumentException("Type with constant $type not found.");
+                    }
+                }
+
+                $currentTag = null;
```

Comment:
if i understand correctly, this could also be at the top of the function, it is to reset the value set by the other function, right? if we do early return we would need to move it up.

Reply:
yes as we are not accessing it inside this method we can set `$currentTag = null;` at beginning